// ng2-collapse/build/src/ng2-collapse/ng2-collapse.component.js
"use strict";
var core_1 = require('@angular/core');
var Ng2CollapseComponent = (function () {
    function Ng2CollapseComponent() {
        this.collapsed = false;
        this.collapse = new core_1.EventEmitter();
        this.expand = new core_1.EventEmitter();
    }
    Ng2CollapseComponent.prototype.ngOnChanges = function (changes) {
        if (changes['collapsed']) {
            this.collapsed ? this.hide() : this.show();
        }
    };
    Ng2CollapseComponent.prototype.toggle = function () {
        this.collapsed = !this.collapsed;
        this.collapsed ? this.hide() : this.show();
    };
    Ng2CollapseComponent.prototype.hide = function () {
        this.collapse.emit({
            collapsed: this.collapsed,
        });
    };
    Ng2CollapseComponent.prototype.show = function () {
        this.expand.emit({
            collapsed: this.collapsed,
        });
    };
    __decorate([
        core_1.Input(), 
        __metadata('design:type', Boolean)
    ], Ng2CollapseComponent.prototype, "collapsed", void 0);
    __decorate([
        core_1.Input(), 
        __metadata('design:type', String)
    ], Ng2CollapseComponent.prototype, "title", void 0);
    __decorate([
        core_1.Output(), 
        __metadata('design:type', Object)
    ], Ng2CollapseComponent.prototype, "collapse", void 0);
    __decorate([
        core_1.Output(), 
        __metadata('design:type', Object)
    ], Ng2CollapseComponent.prototype, "expand", void 0);
    Ng2CollapseComponent = __decorate([
        core_1.Component({
            selector: 'ng2-collapse',
            styles: [require('./ng2-collapse.component.scss')],
            template: require('./ng2-collapse.component.html'),
        }), 
        __metadata('design:paramtypes', [])
    ], Ng2CollapseComponent);
    return Ng2CollapseComponent;
}());
exports.Ng2CollapseComponent = Ng2CollapseComponent;
